build/gen_stub.php

```diff
@@ -1591,24 +1603,56 @@ function (Expr $expr) use ($allConstInfos, &$constType, &$originatingConst, &$is
                 }
 
                 if ($originatingConstName->isUnknown()) {
-                    $originatingConst = null;
-                    $isUnknownConstValue = true;
+                    return;
+                }
+
+                foreach ($this->allConstInfos as $const) {
+                    if ($originatingConstName->equals($const->name)) {
+                        $this->visitedConstants[] = $const;
+                        return $const->getValue($this->allConstInfos)->expr;
+                    }
+                }
+            }
+        };
+
+        $nodeTraverser = new PhpParser\NodeTraverser;
+        $nodeTraverser->addVisitor($visitor);
+        $expr = $nodeTraverser->traverse([$expr])[0];
+
+        $isUnknownConstValue = false;
 
+        $evaluator = new ConstExprEvaluator(
+            function (Expr $expr) use ($allConstInfos, &$isUnknownConstValue) {
+                if (!$expr instanceof Expr\ConstFetch) {
```

Comment:
No, It's been transformed by the visitor above into ConstFetch, always (into the C const, so that the prettyprinter will just give you the right expression).